### PR TITLE
Add Slack integration for error messages

### PR DESCRIPTION
**Pull request by jyotman**
Solves #1 

I've added a `slack` configuration option to *config.js* which has 2 properties - 
1. channelUrl - The webhook's URL for any channel
2. notifyMembers - Users to mention explicitly while posting the message. It can be empty.

As decided I've used the official [slack](https://www.npmjs.com/package/@slack/client) package. A new module has been created as *slack.js*. It provides methods to format and send messages .

Also I've made `smtp` and `slack` configuration options optional as the user may want to use any one of them instead of both.

I've used `let`, `const` and `=>` arror functions which are available in Node.js v4. Let me know if any of those features could be a problem. There's a slight inconsistency currently though due to usage of both `var` and `const/let`. Maybe we can figure that out.

Here's how it would look on Slack - 
slack-demo

Finally I've made the necessary changes to the documentation as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,7 +88,8 @@ Modify the configuration file according to your replica set, especially the foll
 * `database` - the name of your main database
 * `members` - the hostnames of all of your replica set members
 * `auth` - the database authentication username/password
-* `smtp` - the SMTP e-mail configuration for sending alerts (using Gmail, AWS SES, etc)
+* `smtp` (optional) - the SMTP e-mail configuration for sending alerts (using Gmail, AWS SES, etc)
+* `slack` (optional) - the Slack webhook configuration for sending alert messages.
```
Comment (eladnava) [inline README.md]: Please remove the trailing '.' as the other entries don't have it 👍

File: util/logger.js [hunk 2]
```diff
@@ -8,21 +11,35 @@ exports.notifyError = function (error, config, subject) {
...
 
-    // Generate e-mail using mailgen
-    var mail = mailer.generate({
-        body: {
-            title: subject,
-            intro: error.message,
-            outro: 'We thank you for choosing mongomonitor.'
-        }
-    });
+    // Email the error of smt settings present
```
Comment (eladnava) [inline util/logger.js]: Typo: smt

File: util/logger.js [hunk 3]
```diff
@@ -8,21 +11,35 @@ exports.notifyError = function (error, config, subject) {
...
-        from: config.smtp.address,
-        to: config.smtp.address,
-        subject: `[mongomonitor] ${subject}`,
-        html: mail.html,
-        text: mail.text
-    }, config);
+    // Send Slack message if webhook settings present
+    if (config.slack !== undefined) {
+        if (slack === undefined)
+            slack = new Slack(config.slack);
```
Comment (eladnava) [inline util/logger.js]: Please add `{` and `}` brackets to the `if` statement to conform with the rest of the project.

File: util/logger.js [hunk 4]
```diff
@@ -8,21 +11,35 @@ exports.notifyError = function (error, config, subject) {
...
+        slack.sendWebhookMessage(subject, error)
+            .then()
```
Comment (eladnava) [inline util/logger.js]: Empty `.then()` can be omitted. 👍

File: util/slack.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * @param {String} subject
+ * @param {Error} error
+ * @return {Promise}
+ */
+Slack.prototype.sendWebhookMessage = function (subject, error) {
+    const self = this;
+    return new Promise(function (resolve, reject) {
+        const messageBody = self.generateMessage(subject, error);
+        self.webhook.send(messageBody, function (err, res) {
+            if (err)
```
Comment (eladnava) [inline util/slack.js]: Please add brackets. 👍

**Review by eladnava — COMMENTED** [review]
@jyotman Excellent PR!

Had a few comments, please fix and then we should be good to merge! 👍

**Comment by jyotman** [comment]
@eladnava Thanks for reviewing. I've made the requested changes. Please review again.

**Comment by eladnava** [comment]
@jyotman Looks good to me! Merged! 👍  💯

**Comment by eladnava** [comment]
Published in `1.0.2`.

**Pull request merged**